### PR TITLE
feat: polyfill `node:constants` module

### PR DESCRIPTION
**Pull request by jculvey**
Replaces the current auto-mocking of 'constants' to support destructured ESM imports and allow for functional polyfill coverage in the future.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Polyfills the `node:constants` module. Also relocates the `os` constants to this new module and re-exports them from `os`.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/node/constants/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Mostly taken from: https://github.com/nodejs/node/blob/main/typings/internalBinding/constants.d.ts
+
+import constants from "node:constants";
```
Comment (IgorMinar) [inline src/runtime/node/constants/index.ts:3]: did you mean `type`?

```suggestion
import type constants from "node:constants";
```

**Review by IgorMinar — APPROVED** [review]
This looks reasonable to me assuming these values match the real implementation. Could you verify the match against the api scan in the matrix app with values serialized as string/number literals?

File: src/runtime/node/constants/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+export const OPENSSL_VERSION_NUMBER = 269_488_319;
```
Comment (pi0) [inline src/runtime/node/constants/crypto.ts:1]: Honestly, i'm not sure about these hardcoded versions. 

If any library depends on them for a minimal version for specific behavior, we are kinda giving false information to it that _might_ end-up into a chain of issues or possible vulnerabilities. 

I would set SSL versions to `0` to reduce such risks

Comment (IgorMinar) [reply src/runtime/node/constants/crypto.ts:1]: Good catch @pi0. `0` sounds reasonable to me.

**Review by IgorMinar — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**